server/registry/internal/storage/errors.go

```diff
@@ -15,15 +15,78 @@
 package storage
 
 import (
+	"context"
+	"net"
+	"os"
+
+	"github.com/apigee/registry/log"
+	"github.com/lib/pq"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 )
 
+func alreadyExists(err error) bool {
+	// handle sqlite3 errors separately so their support can be conditionally compiled.
+	if isSQLite3ErrorAlreadyExists(err) {
+		return true
+	}
+	switch v := err.(type) {
+	case *pq.Error:
+		if v.Code.Name() == "unique_violation" {
+			return true
+		}
+	}
+	return false
+}
+
 // grpcErrorForDBError converts recognized database error codes to grpc error codes.
-func grpcErrorForDBError(err error) error {
-	if alreadyExists(err) {
-		return status.Error(codes.AlreadyExists, err.Error())
+func grpcErrorForDBError(ctx context.Context, err error) error {
```

Comment:
Would there be any benefit in logging the original error here?

Reply:
On further inspection, I think that only the `unique_violation` err would probably have any additional useful information. Is detail getting logged somewhere?